### DISCUSSION TITLE
If filament now can support reflection an mode from a ground plane ?

### DISCUSSION THREAD

**Discussion by fanbinnet** [google/filament#4518] [Q&A] [answered]
I have read older discussions like:https://github.com/google/filament/discussions/3402
But I think what I need is SSOR like article http://roar11.com/2015/07/screen-space-glossy-reflections/ but not the planar reflections.

I need like below:
![ssgr1](https://user-images.githubusercontent.com/3060688/129840689-953ad338-6b95-4af7-a030-c82f5e06631a.JPG)

In other issues I notice filament can support screen-space reflections .
But I can just implement cubemap reflection, the model don't reflection like below.

![ssor2](https://user-images.githubusercontent.com/3060688/129840694-82d315c7-15cf-46fb-9bb0-b4dcdbec0171.JPG)

The material have been set to:
  refractionType : solid,
  refractionMode : screenspace,
and in code :
        view.setShadowingEnabled(false)
        view.setScreenSpaceRefractionEnabled(true)

**Comment by romainguy** [comment, marked as answer]
Filament suports screen space refraction, not reflections. Screen space reflections are being implemented though. If you want to do planar reflections you can check this sample: https://github.com/google/filament/blob/main/samples/rendertarget.cpp

**Reply by fanbinnet** [reply]
Thanks for you reply
